File: example/matlab/snesim_example.m
% load example training image ('strebelle')
ti=imread('https://raw.githubusercontent.com/GAIA-UNIL/TrainingImagesTIFF/master/strebelle.tiff');

% SNESIM call using G2S
simulation=g2s('-a','snesim',...
               '-ti',ti,...
               '-di',nan(1000,1000),...
               '-dt',[1],...
               '-j',0.5,...
               '-mg',4,...
               '-tpl',3);

% Display results
sgtitle('SNESIM unconditional simulation');
subplot(1,2,1);
imshow(ti,[]);
title('Training image');
subplot(1,2,2);
imshow(simulation,[]);
title('Simulation');

File: example/python/snesim/snesim_example.py
import numpy as np
from PIL import Image
import requests
from io import BytesIO
from g2s import g2s


ti = np.array(Image.open(BytesIO(requests.get(
    "https://raw.githubusercontent.com/GAIA-UNIL/TrainingImagesTIFF/master/Strebelle.tiff"
).content)), dtype=np.float32)

result = g2s(
    "-a", "snesim",
    "-ti", ti,
    "-di", np.full(ti.shape, np.nan, dtype=np.float32),
    "-dt", [1],
    "-n", 24,
    "-s", 100,
)

print("job", result["job_id"], "seconds", result.get("time"))
print("simulation", result["simulation"].shape)
